sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -255,41 +261,145 @@ case class HashAggregateExec(
      """.stripMargin
   }
 
+  private def isValidParamLength(paramLength: Int): Boolean = {
+    sqlContext.getConf("spark.sql.HashAggregateExec.isValidParamLength", null) match {
```

Comment:
Will we define valid parameter length specifically for hash aggregate?

Reply:
ur, i forgot to add a comment here "only for testing"...